src/app/widget/widget.component.ts

```diff
@@ -1,49 +1,68 @@
 import {Component, OnInit} from '@angular/core';
-import {WeatherService} from "../services/weather.service";
-import {interval, startWith} from 'rxjs';
-import {map,switchMap} from 'rxjs/operators';
 import {WeatherWidget} from "../interfaces/weatherwidget";
 import {WidgetUiMode} from "./WidgetUiMode";
+import {LocalStorageService} from "../services/loacalstorageservice/localstorage.service";
+import {WidgetService} from "../services/widgetservice/widget.service";
 
 @Component({
-  selector: 'app-widget',
-  templateUrl: './widget.component.html',
-  styleUrls: ['./widget.component.scss']
+  selector: 'app-widget', templateUrl: './widget.component.html', styleUrls: ['./widget.component.scss']
 })
 export class WidgetComponent implements OnInit {
   weatherWidgets: WeatherWidget[] = [
     new WidgetUiMode({} as WeatherWidget),
     new WidgetUiMode({} as WeatherWidget),
     new WidgetUiMode({} as WeatherWidget),
   ];
-  btnClass: string[] = [];
-  constructor(private weatherService: WeatherService,) {}
+
+  constructor(
+    private storageService: LocalStorageService,
+    private widgetService: WidgetService,) {
+  }
+
   ngOnInit() {
+    this.localStorage();
+    this.updateWeather();
+  }
+
+  updateWeather() {
+    this.weatherWidgets.forEach((value) => {
+      this.getWeather(value.id);
+    })
   }
-  showInput(index: number) {
-    const widget = this.weatherWidgets[index];
+
+  showInput(id: number) {
+    const widget = this.weatherWidgets.find((item) => {
+      return item.id === id;
+    }) || {} as WidgetUiMode;
     widget.flag = false;
-    this.btnClass[index] = 'display_none';
-  }
-
-  getWeather(index: number) {
-    const widget = this.weatherWidgets[index];
-    interval(3000)
-      .pipe(
-        startWith(0),
-        switchMap(() => this.weatherService.getWeather(widget.name)),
-        map((data) => new WidgetUiMode(data))
-      )
-      .subscribe(data => {
-        widget.weatherData = data;
-        widget.name = data.name;
-        widget.main.temp = data.main.temp;
-        widget.main.temp_min = data.main.temp_min;
-        widget.main.temp_max = data.main.temp_max;
-        widget.flag = true;
-
-        console.log(data);
-      });
+  }
+
+  getWeather(id: number) {
+    const widget = this.weatherWidgets.find((item) => {
+      return item.id === id;
+    }) || {} as WidgetUiMode;
+    this.widgetService.serviceData(widget)
+    this.setLocalStorage();
+  }
+
```

Comment:
runWatcher() {
  **interval(3000).pipe(
    switchMap(() => this.weatherService.serviceData()),
        map((data) => new WidgetUiMode(data))**.subscribe(data => widget = data)
      )
}